gradio/ranged_response.py

```diff
@@ -0,0 +1,185 @@
+# Taken from https://gist.github.com/kevinastone/a6a62db57577b3f24e8a6865ed311463
+# Context: https://github.com/encode/starlette/pull/1090
+
+import os
+import re
+import stat
+import typing as t
```

Comment:
To be consistent with other modules, I would just do `from typing import NamedTuple`, etc. 

We also don't use `typing.Union` and instead use `|`

Reply:
I didn't edit this file because it's copy-pasted from the github link at the top of the file - it's almost like a "third-party" library. Would think it's better not to touch third party code?

Reply:
have moved to folder called "third_party", which is ignored by style and codecov tests.

Reply:
It isn't third-party if we bring it into the repo manually, so it meeds meet the same code standards as the rest of the code-base, since we need to maintain it.

Reply:
hm okay if that's standard practice, that makes sense

Reply:
fixed